repositories/productRepository.go

```diff
@@ -13,67 +16,173 @@ type ProductRepository struct {
+		}
+	}
+	finalQuery.WriteString(";")
+
+	products := []models.Product{}
+
+	rowsProducts, err := repository.DB.Query(context.Background(), finalQuery.String(), employeeIdc)
+	defer rowsProducts.Close()
+	if err != nil {
+		return nil, err
+	}
+	for rowsProducts.Next() {
 		var product models.Product
-		err := rows.Scan(&product.ID, &product.Name, &product.Price, &product.Quantity, &product.IDC)
+		err := rowsProducts.Scan(&product.ID, &product.Name, &product.Price, &product.Quantity, &product.IDC)
 		if err != nil {
 			return nil, err
 		}
 		products = append(products, product)
 	}
+
 	return products, nil
 }
 
 func (repository *ProductRepository) GetProduct(id string) (models.Product, error) {
 	var product models.Product
-	err := repository.DB.QueryRow(context.Background(),
-		"SELECT * FROM products WHERE id=$1", id).
-		Scan(&product.ID, &product.Name, &product.Price, &product.Quantity, &product.IDC)
+
+	Uuid, err := uuid.FromString(id)
 	if err != nil {
 		return product, err
 	}
+
+	rows, err := repository.DB.Query(context.Background(), "select * from products where id=$1", Uuid)
+	defer rows.Close()
+
+	if err != nil {
+		return product, err
+	}
+
+	if !rows.Next() {
+		return product, errors.New("There is no product with this id")
+	}
+
+	var productPers persistence.Products
+	productPers.Scan(&rows)
+
+	var productUUID string
+	err = productPers.Id.AssignTo(&productUUID)
+	if err != nil {
+		return product, err
+	}
+
+	var companyUUID string
+	err = productPers.Idc.AssignTo(&companyUUID)
+	if err != nil {
+		return product, err
+	}
+
+	product = models.Product{
+		ID:       productUUID,
+		Name:     productPers.Name,
+		Price:    productPers.Price,
+		Quantity: productPers.Quantity,
+		IDC:      companyUUID,
+	}
+
 	return product, nil
 }
 
 func (repository *ProductRepository) AddProduct(product *models.Product) error {
-	u := uuid.NewV4()
-	product.ID = u.String()
-	_, err := repository.DB.Exec(context.Background(),
-		"INSERT INTO products (id, name, price, quantity, idc) values ($1, $2, $3, $4, $5)",
-		u.Bytes(), product.Name, product.Price, product.Quantity, product.IDC)
+	tx, err := repository.DB.Begin(context.Background())
+	if err != nil {
+		return err
+	}
+	defer tx.Rollback(context.Background())
+
+	product.ID = uuid.NewV4().String()
+
+	productPers := persistence.Products{
+		Name:     product.Name,
+		Price:    product.Price,
+		Quantity: product.Quantity,
+	}
+	productPers.Idc.Set(product.IDC)
+	productPers.Id.Set(product.ID)
+
+	_, err = productPers.InsertTx(&tx)
 	if err != nil {
 		return err
 	}
-	return nil
+
+	return tx.Commit(context.Background())
 }
 
 func (repository *ProductRepository) UpdateProduct(product models.Product) error {
-	commandTag, err := repository.DB.Exec(context.Background(),
-		"UPDATE products SET name=$1, price=$2, quantity=$3, idc=$4 WHERE id=$5",
-		product.Name, product.Price, product.Quantity, product.IDC, product.ID)
+	tx, err := repository.DB.Begin(context.Background())
```

Comment:
ExternalRights check missing here and in DeleteProduct.
What if an employee from org A wants to perform update or delete on a product belonging to org B?

Reply:
We did permision check in repository only for GetAll functions, all other permission checks are in service, is it ok to stay like that

Reply:
My bad sorry, didn't saw it. It looks ok like that. You just have a typo here in the service Updateproduct method.